File: app/src/main/java/com/kent/university/privelt/ui/login/ScriptsAdapter.java
package com.kent.university.privelt.ui.login;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.kent.university.privelt.R;
import com.kent.university.privelt.model.Service;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class ScriptsAdapter extends RecyclerView.Adapter<ScriptViewHolder> {

    private LinkedHashMap<String, Boolean> scripts;

    public ScriptsAdapter(List<String> scripts, List<String> alreadyChecked) {

        this.scripts = new LinkedHashMap<>();

        for (String script : scripts) {
            this.scripts.put(script, false);
        }
        for (String script : alreadyChecked) {
            Log.d("TAG", script);
            if (!script.isEmpty())
                this.scripts.put(script, true);
        }
    }

    @NonNull
    @Override
    public ScriptViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.cell_script, parent, false);
        return new ScriptViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ScriptViewHolder holder, int position) {
        holder.bind((String) scripts.keySet().toArray()[position], scripts);
    }

    @Override
    public int getItemCount() {
        return scripts.size();
    }

    public String getConcatenatedScriptsChecked() {
        String result = "";
        for (Map.Entry<String, Boolean> entry : scripts.entrySet()) {
            String key = entry.getKey();
            Boolean value = entry.getValue();
            if (value)
                result = result.concat(key.concat(Service.DELIMITER));
        }
        return result;
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/login/ScriptsAdapter.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.login

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.kent.university.privelt.R
import com.kent.university.privelt.model.Service
import java.util.*

class ScriptsAdapter internal constructor(scripts: List<String>, alreadyChecked: List<String>) : RecyclerView.Adapter<ScriptViewHolder>() {
    private val scripts: LinkedHashMap<String?, Boolean> = LinkedHashMap()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScriptViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.cell_script, parent, false)
        return ScriptViewHolder(view)
    }

    override fun onBindViewHolder(holder: ScriptViewHolder, position: Int) {
        holder.bind(scripts.keys.toTypedArray()[position], scripts)
    }

    override fun getItemCount(): Int {
        return scripts.size
    }

    val concatenatedScriptsChecked: String
        get() {
            var result = ""
            for ((key, value) in scripts) {
                if (value) result = result + key + Service.DELIMITER
            }
            return result
        }

    init {
        for (script in scripts) {
            this.scripts[script] = false
        }
        for (script in alreadyChecked) {
            if (script.isNotEmpty()) this.scripts[script] = true
        }
    }
}